framework/src/main/java/org/checkerframework/framework/type/AnnotatedTypeMirror.java

```diff
@@ -154,6 +157,8 @@ private AnnotatedTypeMirror(TypeMirror underlyingType, AnnotatedTypeFactory atyp
         this.underlyingType = underlyingType;
         assert atypeFactory != null;
         this.atypeFactory = atypeFactory;
+        this.underlyingTypeHashcode = -1;
```

Comment:
I would put this at the field declaration, making it explicit what the "uninit" value is.  (Currently that is buried in the constructor.)